feat: transaction status subscribing

kuba-4chain opened the pull request:
## Description of Changes

This change allows users of wallet-toolbox to subscribe for updates when a transaction is broadcasted and when it's mined (proven).

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [x] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run `npm run doc` and `npm run lint` one final time before requesting a review
- [x] I have run `npm version patch` so that my changes will trigger a new version to be released when they are merged

tonesnotes commented:
And a naming issue: "process" has specific meaning in the toolbox code: All the existing "process" functions are involved in putting the pieces of a transaction together and turning them into a valid bitcoin transaction.

How about "callOnProvenTransaction" and "callOnBroadcastedTransaction"?

sirdeggen commented on src/monitor/Monitor.ts:
```diff
@@ -286,6 +296,34 @@ export class Monitor {
+   */
+  processBroadcastedTransaction(broadcastResult: sdk.ReviewActionResult): void {
+    if (this.onTransactionBroadcasted) {
+      console.log('Sending transaction broadcast update up to the toolbox client')
```
```suggestion
     // console.log('Sending transaction broadcast update up to the toolbox client')
```

tonesnotes reviewed (changes requested):
Changes have been requested.

kuba-4chain commented:
@tonesnotes good points, I've pushed the requested changes

1. I wanted to make this feature as generic as possible for future users, but for our use case the `rawTx` is not needed. And you're right that it's passed by reference and there are other ways to get it. Removed.
2. Good catch, I've pulled the implementation of the aggregateResult function and changed it to only be invoked if a client is subscribed. Let me know if that's more or less what you had in mind.
3. Naming is always a tricky thing. Changed to `onCall...`

kuba-4chain commented on src/monitor/Monitor.ts:
```diff
@@ -286,6 +296,34 @@ export class Monitor {
+   */
+  processProvenTransaction(txStatus: sdk.ProvenTransactionStatus): void {
+    if (this.onTransactionProven) {
+      console.log('Sending proven transaction update up to the toolbox client')
```
Removed
